assets/src/containers/AssignmentPlanning.js

```diff
@@ -133,82 +135,85 @@ function AssignmentPlanning (props) {
   if (assignmentError) return (<WarningBanner />)
 
   return (
-    <div className={classes.root}>
-      <Grid container spacing={2}>
-        <Grid item xs={12}>
-          <Paper className={classes.paper}>
-            <ViewHeader>Assignment Planning</ViewHeader>
-            <div className={classes.section}>
-              <Typography variant='h6' gutterBottom>Progress toward Final Grade</Typography>
-              {
-                assignmentData
-                  ? (
-                    <ProgressBar
-                      data={assignmentData.progress}
-                      aspectRatio={0.12}
-                      tip={AssignmentPlanningTooltip(classes)}
-                    />
-                  )
-                  : <Spinner />
-              }
-            </div>
-            <div className={classes.section}>
-              <Grid container>
-                <Grid item xs={12} md={10}>
-                  <Typography variant='h6' gutterBottom>Assignments Due by Date</Typography>
+    <>
+      <PreviewBanner isDisabled={disabled} />
+      <div className={classes.root}>
```

Comment:
Looks like you needed to create this external `<>` so you could put the `Banner` on the same level as the `root` `div`. I'm not sure (but could be convinced) about this decision. I think it might look cleaner if the `Banner` was on the main `Paper` with the rest of the content -- the width would then be the same as the other `Banner` indicating no data. We could even consider putting this banner below the View title.

Less important side note: Another advantage of this would be that you'd technically be editing many fewer lines of code. If it's really necessary, that's fine, but it does end up obscuring history, if you want to know when/why a particular part of the component was changed in the past.

Reply:
I liked the idea of the banner effectively existing outside of the view so it would be a true preview. I hadn't considered the blame impact. I care more about the structure of the code than making it easier to go through history, but I don't feel strongly about either aspect in this case, though I'd tend to leave it how it is currently..

Reply:
Okay, I think you convinced me. Might me worth checking with @jennlove-um on the appearance, but I'm fine with this.